### Heroes Of Pymoli Data Analysis


In [1]:
# Dependencies and Setup
import pandas as pd
import numpy as np

# File to Load (Remember to Change These)
file_to_load = "Resources/purchase_data.csv"

# Read Purchasing File and store into Pandas data frame
purchase_data = pd.read_csv(file_to_load)

## Player Count

* Display the total number of players


In [2]:
print("Total Number of player is "+ str(len(purchase_data['SN'].unique())))

Total Number of player is 576


## Purchasing Analysis (Total)

* Run basic calculations to obtain number of unique items, average price, etc.


* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame

In [3]:
# Number of unique itemb
unq_item = len(purchase_data['Item ID'].unique())

# Average Price
avg_price = round(purchase_data['Price'].mean(),2)

# no of purchases
purchase = len(purchase_data['Purchase ID'].unique())

# Total Revinue
tot_rev = round(purchase_data['Price'].sum(),2)

In [4]:
# Make the dataframe
summarydf = pd.DataFrame({"Number of Unique Items":[unq_item],"Average Price":[avg_price],"Number of Purchases":[purchase],"Total Revenue":[tot_rev]} )

In [5]:
# Correct formatting
summarydf['Total Revenue'] = "$" +  summarydf['Total Revenue'].astype(str)

In [6]:
summarydf

Number of Unique Items  Average Price  Number of Purchases Total Revenue
0                     183           3.05                  780      $2379.77

## Gender Demographics

* Percentage and Count of Male Players


* Percentage and Count of Female Players


* Percentage and Count of Other / Non-Disclosed




In [7]:
# Calculate count and percentages
gender_demo = purchase_data[['SN','Gender']].groupby(['Gender']).count()
gender_demo['Percent of Players'] = (gender_demo.SN / gender_demo.SN.sum())*100
gender_demo = gender_demo[['Percent of Players']]
gender_demo

Percent of Players
Gender                                   
Female                          14.487179
Male                            83.589744
Other / Non-Disclosed            1.923077


## Purchasing Analysis (Gender)

* Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. by gender




* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame

In [8]:
# Calculate purchase count
purchase_count = purchase_data[['Purchase ID','Gender']].groupby(['Gender']).count().rename(columns =   
                                                                                            {'Purchase ID':'Purchase Count'})

In [9]:
# Calculate avg price per person
avg_price = purchase_data[['Price','Gender']].groupby(['Gender']).mean().rename(columns =
                                                                                            {'Price':'Average Purchase Price'})

In [10]:
# Calcualte total purchase
total_purchase = purchase_data[['Price','Gender']].groupby(['Gender']).sum().rename(columns =
                                                                                            {'Price':'Total Purchase Value'})


In [11]:
td = purchase_data.groupby(['SN','Gender']).sum()

In [12]:
# calculate total purchase
td = td.groupby(['Gender']).mean().rename(columns = {'Price':'Avg Total Purchase per Person'})
td = td[['Avg Total Purchase per Person']]

In [13]:
# Combine the data
purchase_analysis = purchase_count
purchase_analysis['Average Purchase Price'] ="$"+  avg_price[['Average Purchase Price']].round(2).astype(str)
purchase_analysis['Total Purchase Value'] = "$"+ total_purchase[['Total Purchase Value']].round(2).astype(str)
purchase_analysis['Avg Total Purchase per Person'] ="$"+  td[['Avg Total Purchase per Person']].round(2).astype(str)
purchase_analysis

Purchase Count Average Purchase Price  \
Gender                                                         
Female                            113                   $3.2   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Value Avg Total Purchase per Person  
Gender                                                                    
Female                             $361.94                         $4.47  
Male                              $1967.64                         $4.07  
Other / Non-Disclosed               $50.19                         $4.56

## Age Demographics

* Establish bins for ages


* Categorize the existing players using the age bins. Hint: use pd.cut()


* Calculate the numbers and percentages by age group


* Create a summary data frame to hold the results


* Optional: round the percentage column to two decimal points


* Display Age Demographics Table


In [14]:
# * Establish bins for ages
age_bins = [0, 9, 14, 19, 24, 29, 34, 39,45]
group_names = ["<10", "10-14", "15-19","20-24","25-29","30-34","35-39","40+"]

demo_data = purchase_data


# * Categorize the existing players using the age bins. Hint: use pd.cut()
demo_data['Age Group'] = pd.cut(purchase_data['Age'], bins=age_bins, labels = group_names)

# * Calculate the numbers and percentages by age group
demo_data = demo_data.groupby(['Age Group']).count().rename(columns = {'Purchase ID':'Total Count'})
demo_data['Percentage of Players'] = (demo_data['Total Count'].values / demo_data['Total Count'].sum())*100

In [15]:
# * Create a summary data frame to hold the results
summary_df = demo_data[['Total Count','Percentage of Players']]

In [16]:
# * Optional: round the percentage column to two decimal points

In [17]:
# Correct formatting
summary_df['Percentage of Players'] = summary_df['Percentage of Players'].round(2)

C:\ProgramData\Anaconda3\lib\site-packages\ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


In [18]:
# * Display Age Demographics Table
summary_df

Total Count  Percentage of Players
Age Group                                    
<10                 23                   2.95
10-14               28                   3.59
15-19              136                  17.44
20-24              365                  46.79
25-29              101                  12.95
30-34               73                   9.36
35-39               41                   5.26
40+                 13                   1.67

## Purchasing Analysis (Age)

* Bin the purchase_data data frame by age


* Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. in the table below


* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame

In [19]:
#Store age seperately
analysis_data = purchase_data

# * Bin the purchase_data data frame by age

age_bins = [0, 9, 14, 19, 24, 29, 34, 39,45]
group_names = ["<10", "10-14", "15-19","20-24","25-29","30-34","35-39","40+"]
# * Establish bins for ages
purchase_data['Age Group'] = pd.cut(purchase_data['Age'], bins=age_bins, labels = group_names)

In [20]:
# * Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. in the table below
count = purchase_data.groupby(['Age Group']).count()
count = count.rename(columns={'Purchase ID':'Purchase Count'})
count = count[['Purchase Count']]

In [21]:
# Calcualte average price
avg_price = purchase_data[['Price','Age Group']].groupby(['Age Group']).mean().rename(columns =
                                                                                            {'Price':'Average Purchase Price'})

In [22]:
# calculate total price
total_purchase = purchase_data[['Price','Age Group']].groupby(['Age Group']).sum().rename(columns =
                                                                                            {'Price':'Total Purchase Value'})

In [23]:
# calculate total
td = purchase_data.groupby(['SN','Age Group']).sum()

In [24]:
td.reset_index(inplace= True)

In [25]:
td = td.groupby(['Age Group']).mean().rename(columns = {'Price':'Avg Total Purchase per Person'})
td = td[['Avg Total Purchase per Person']]


In [26]:
# * Create a summary data frame to hold the results
purchase_analysis = count
purchase_analysis['Average Purchase Price'] = avg_price['Average Purchase Price']
purchase_analysis['Total Purchase Value'] = total_purchase['Total Purchase Value']
purchase_analysis['Avg Total Purchase per Person'] = td['Avg Total Purchase per Person']


In [27]:
# * Optional: give the displayed data cleaner formatting

In [28]:
purchase_analysis['Average Purchase Price'] = "$"+ purchase_analysis['Average Purchase Price'].round(2).astype(str)
purchase_analysis['Total Purchase Value'] = "$"+ purchase_analysis['Total Purchase Value'].round(2).astype(str)
purchase_analysis['Avg Total Purchase per Person'] = "$"+ purchase_analysis['Avg Total Purchase per Person'].round(2).astype(str)

In [29]:
# * Display the summary data frame
purchase_analysis

Purchase Count Average Purchase Price Total Purchase Value  \
Age Group                                                               
<10                    23                  $3.35               $77.13   
10-14                  28                  $2.96               $82.78   
15-19                 136                  $3.04              $412.89   
20-24                 365                  $3.05             $1114.06   
25-29                 101                   $2.9               $293.0   
30-34                  73                  $2.93               $214.0   
35-39                  41                   $3.6              $147.67   
40+                    13                  $2.94               $38.24   

          Avg Total Purchase per Person  
Age Group                                
<10                               $4.54  
10-14                             $3.76  
15-19                             $3.86  
20-24                             $4.32  
25-29                             $3.81  
30-34                             $4.12  
35-39                             $4.76  
40+                               $3.19

## Top Spenders

* Run basic calculations to obtain the results in the table below


* Create a summary data frame to hold the results


* Sort the total purchase value column in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the summary data frame



In [30]:
# * Run basic calculations to obtain the results in the table below
purchase_count = purchase_data.groupby(['SN']).count().rename(columns = {'Purchase ID':"Purchase Count"})
purchase_count = purchase_count[['Purchase Count']]

In [31]:
# calcualte average purchase price
avg_price = purchase_data.groupby(['SN']).mean().rename(columns = {'Price':"Average Purchase Price"})
avg_price = avg_price[['Average Purchase Price']]

In [32]:
# Calcualte total purchase value
tot_price = purchase_data.groupby(['SN']).sum().rename(columns = {'Price':"Total Purchase Value"})
tot_price = tot_price[['Total Purchase Value']]

In [33]:
# * Create a summary data frame to hold the results
top_spenders = purchase_count
top_spenders['Average Purchase Price'] = avg_price['Average Purchase Price']
top_spenders['Total Purchase Value'] = tot_price['Total Purchase Value']


In [34]:
# * Sort the total purchase value column in descending order
top_spenders = top_spenders.sort_values('Total Purchase Value', ascending= False)

In [35]:
# * Optional: give the displayed data cleaner formatting
top_spenders['Average Purchase Price'] = "$"+ top_spenders['Average Purchase Price'].round(2).astype(str)
top_spenders['Total Purchase Value'] = "$"+ top_spenders['Total Purchase Value'].round(2).astype(str)

In [36]:
# * Display a preview of the summary data frame
top_spenders.head()

Purchase Count Average Purchase Price Total Purchase Value
SN                                                                     
Lisosia93                 5                  $3.79               $18.96
Idastidru52               4                  $3.86               $15.45
Chamjask73                3                  $4.61               $13.83
Iral74                    4                   $3.4               $13.62
Iskadarya95               3                  $4.37                $13.1

## Most Popular Items

* Retrieve the Item ID, Item Name, and Item Price columns


* Group by Item ID and Item Name. Perform calculations to obtain purchase count, item price, and total purchase value


* Create a summary data frame to hold the results


* Sort the purchase count column in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the summary data frame



In [37]:
# * Retrieve the Item ID, Item Name, and Item Price columns
pop_data = purchase_data[['Item ID','Item Name','Price']]


In [38]:
# * Group by Item ID and Item Name. Perform calculations to obtain purchase count, item price, and total purchase value
pop_data_items = pop_data.groupby(['Item ID','Item Name']).count().rename(columns={'Price':'Purchase Count'})

In [39]:
# Take Item Price
pop_data_price = pop_data.groupby(['Item ID','Item Name','Price']).count()
pop_data_price = pop_data_price.reset_index()
pop_data_price = pop_data_price.rename(columns = {'Price':'Item Price'})

In [40]:
# Calculate total price
tot_price = pop_data.groupby(['Item ID','Item Name']).sum()
tot_price.reset_index(inplace=True)
tot_price['Total Purchase Value'] = tot_price['Price']
tot_price = tot_price[['Item ID','Item Name','Total Purchase Value']]

In [41]:
pop_data_items.reset_index(inplace=True)
pop_data_price.reset_index(inplace=True)
tot_price.reset_index(inplace = True)

In [42]:
# Sort 
pop_data_items = pop_data_items.sort_values(['Purchase Count'], ascending=False)
# * Create a summary data frame to hold the results

summary_df = pd.merge(pop_data_items,pop_data_price )
summary_df = pd.merge(summary_df, tot_price)


In [43]:
# * Optional: give the displayed data cleaner formatting
summary_df['Item Price'] = "$"+ summary_df['Item Price'].round(2).astype(str)
summary_df['Total Purchase Value num'] = summary_df['Total Purchase Value'].round(2)
summary_df['Total Purchase Value'] = "$"+ summary_df['Total Purchase Value'].round(2).astype(str)

In [44]:
# * Display a preview of the summary data frame
summary_df[['Item Name','Purchase Count', 'index','Item Price','Total Purchase Value']].head()

Item Name  Purchase Count  index  \
0  Oathbreaker, Last Hope of the Breaking Storm              12    177   
1                          Fiery Glass Crusader               9    144   
2     Extraction, Quickblade Of Trembling Hands               9    107   
3                                       Nirvana               9     81   
4                 Pursuit, Cudgel of Necromancy               8     19   

  Item Price Total Purchase Value  
0      $4.23               $50.76  
1      $4.58               $41.22  
2      $3.53               $31.77  
3       $4.9                $44.1  
4      $1.02                $8.16

## Most Profitable Items

* Sort the above table by total purchase value in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the data frame



In [45]:
summary_df = summary_df.sort_values(['Total Purchase Value num'], ascending=False)

In [46]:
summary_df[['Item Name','Purchase Count', 'index','Item Price','Total Purchase Value']].head()

Item Name  Purchase Count  index  \
0   Oathbreaker, Last Hope of the Breaking Storm              12    177   
3                                        Nirvana               9     81   
1                           Fiery Glass Crusader               9    144   
12                                  Final Critic               8     91   
5                                 Singed Scalpel               8    102   

   Item Price Total Purchase Value  
0       $4.23               $50.76  
3        $4.9                $44.1  
1       $4.58               $41.22  
12      $4.88               $39.04  
5       $4.35                $34.8

# Written description of three observable trends based on the data.

* Among 576 Player most are males 83.6 %.Most products are purchased by male, Total Purchase value for male 1967.64 is the highests. So male dominates the game.
* Most of the players are aged between 15 to 29.From the purchase analysis data we can see that player aged between 15 to 29 has the highest total purchase values and highest total average per person purchase value.Here the trend is that 15-29 years players are more active in terms of playing and purchasing products.
* Most popular items and profitable items were Oathbreaker, Last Hope of the Breaking Storm,Nirvana,Fiery Glass Crusader,Final Critic,Singed Scalpel. So here the trend is the more popular the item the more profitable it is.